sklearn/metrics/tests/test_score_objects.py

```diff
@@ -145,37 +151,125 @@ def test_all_scorers_repr():
+
+    for scoring in (('accuracy',), ['precision'],
+                    {'acc': 'accuracy', 'precision': 'precision'},
+                    ('accuracy', 'precision'), ['precision', 'accuracy'],
+                    {'accuracy': make_scorer(accuracy_score),
+                     'precision': make_scorer(precision_score)}):
+        estimator = EstimatorWithFitAndPredict()
+        estimator.fit([[1]], [1])
+
+        scorers, is_multi = check_multimetric_scoring(estimator, scoring)
+        assert_true(is_multi)
+        assert_true(isinstance(scorers, dict))
+        assert_equal(sorted(scorers.keys()), sorted(list(scoring)))
+        assert_true(all([isinstance(scorer, _PredictScorer)
+                         for scorer in list(scorers.values())]))
+
+        if 'acc' in scoring:
+            assert_almost_equal(scorers['acc'](estimator, [[1]], [1]), 1)
+        if 'accuracy' in scoring:
+            assert_almost_equal(scorers['accuracy'](estimator, [[1]], [1]), 1)
+        if 'precision' in scoring:
+            assert_almost_equal(scorers['precision'](estimator, [[1]], [0]), 0)
+
+    estimator = EstimatorWithFitAndPredict()
+    estimator.fit([[1]], [1])
+    scorers, is_multi = check_multimetric_scoring(estimator, allow_none=True)
+    assert_equal(scorers, {'score': None})
+    assert_false(is_multi)
+
+    # Make sure it raises errors when scoring parameter is not valid.
+
+    error_message_regexp = ".*must be unique strings.*\n.*use a dict.*"
+
+    # List/tuple of callables should raise a message advising users to use
+    # dict of names to callables mapping
+    assert_raises_regexp(ValueError, error_message_regexp,
+                         check_multimetric_scoring, estimator,
+                         scoring=(make_scorer(precision_score),
+                                  make_scorer(accuracy_score)))
+
+    assert_raises_regexp(ValueError, error_message_regexp,
+                         check_multimetric_scoring, estimator,
+                         scoring=(make_scorer(precision_score),))
+
+    # So should empty lists/tuples
+    assert_raises_regexp(ValueError, error_message_regexp,
+                         check_multimetric_scoring, estimator, scoring=())
+
+    # So should duplicated entries
+    assert_raises_regexp(ValueError, error_message_regexp,
+                         check_multimetric_scoring, estimator,
+                         scoring=('f1_micro', 'f1_micro'))
+
+    error_message_regexp = (".*should.*be.*string or callable.*for single"
+                            ".*dict.*for multi.*")
+
+    # Empty dict should raise invalid scoring error
+    assert_raises_regexp(ValueError, error_message_regexp,
+                         check_multimetric_scoring, estimator,
+                         scoring=(dict()))
+
+    # And so should any other invalid entry
+    assert_raises_regexp(ValueError, error_message_regexp,
+                         check_multimetric_scoring, estimator, scoring=5)
+
```

Comment:
I suppose you should also check that an appropriate error is raised if one (but not all) of the values is an invalid scorer. In particular, the error message should specify which is invalid.